### PR TITLE
Issue1165

### PR DESCRIPTION
**Pull request by isivigno**
# PR Description

Add postfix ++ to ArithDSSIterator (issue#1165)

# Checklist

- [ ] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [ ] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [ ] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/geometry/curves/FrechetShortcut.h [hunk 1]
```diff
@@ -356,20 +356,18 @@ namespace DGtal
...
 	    return false;
 	else
 	  {
-	    //otherwise, translate the points such that a->0
-	    int tmp = a;
-	    a = fmod(a+n-tmp,n);
-	    b = fmod(b+n-tmp,n);
-	    i = fmod(i+n-tmp,n);
-	    return isBetween(i,a,b,n); 
+	    if((i>=a && i<=n) || (i>=0 && i<=b))
```
Comment (kerautret) [inline src/DGtal/geometry/curves/FrechetShortcut.h]: saving an if with "return (i>=a && i<=n) || (i>=0 && i<=b)"  ?

File: src/DGtal/geometry/curves/ArithDSSIterator.h [hunk 2]
```diff
@@ -110,6 +110,10 @@ namespace DGtal
      */
     Self& operator++();
 
+    Self operator++(int);
```
Comment (kerautret) [inline src/DGtal/geometry/curves/ArithDSSIterator.h]: you forget the oxygen doc ;)

File: src/DGtal/geometry/curves/FrechetShortcut.h [hunk 3]
```diff
@@ -356,20 +356,18 @@ namespace DGtal
       */
       static  bool isBetween(double i, double a, double b, double n)
       {
-	// if a<=b, a simple comparison enables to conclude
 	if(a<=b)
 	  if(i>=a && i<=b)
```
Comment (kerautret) [inline src/DGtal/geometry/curves/FrechetShortcut.h]: here too return (i>=a && i<=b);

**Comment by kerautret** [comment]
@isivigno looks fine just minor remarks, if you want to solve the conflict, you can merge my PR that I put on your remote.

**Comment by dcoeurjo** [comment]
Hi @isivigno , can I review this PR ?

File: ChangeLog.md [hunk 4]
```diff
@@ -114,6 +114,8 @@
    when displaying a small segment. Fix a non initialized attribute with
    some improvements on bounding box computation with orientation check.
    (B. Kerautret, [#1123](https://github.com/DGtal-team/DGtal/pull/1123))
+ - ArithDSSIterator: fix missing postfix ++.
```
Comment (dcoeurjo) [inline ChangeLog.md]: can you please move this block to 0.9.3 release ?

**Comment by dcoeurjo** [comment]
Excellent, thank you  @isivigno

**Pull request merged**